aries-site/src/examples/templates/forms/managing-child-objects/CreateCluster.js

```diff
@@ -0,0 +1,238 @@
+import { useEffect, useState } from 'react';
+import {
+  Box,
+  Button,
+  CheckBox,
+  Form,
+  FormField,
+  Heading,
+  MaskedInput,
+  RangeInput,
+  Select,
+  Text,
+  TextInput,
+} from 'grommet';
+import { FormChildObjects } from '../../../../examples/templates/FormChildObject';
+
+const hostTemplate = {
+  name: '',
+  host: '',
+  cpu: '',
+  memory: '',
+};
+
+const INPUT_MAP = {
+  name: ({ key, index, ...rest }) => (
+    <FormField
+      key={key}
+      htmlFor={`hosts[${index}].name`}
+      name={`hosts[${index}].name`}
+      label="Host name"
+    >
+      <TextInput
+        id={`hosts[${index}].name`}
+        name={`hosts[${index}].name`}
+        {...rest}
+      />
+    </FormField>
+  ),
+  host: ({ key, index, ...rest }) => (
+    <FormField
+      key={key}
+      htmlFor={`hosts[${index}].host`}
+      name={`hosts[${index}].host`}
+      label="Host address"
+      // Nested required fields e.g. hosts[2].host is not being picked up by validation
+      required
+      aria-required="true"
+    >
+      <TextInput
+        id={`hosts[${index}].host`}
+        name={`hosts[${index}].host`}
+        {...rest}
+      />
+    </FormField>
+  ),
+  cpu: ({ key, index, ...rest }) => (
+    <Box key={key} width="xsmall">
+      <FormField
+        htmlFor={`hosts[${index}].cpu`}
+        name={`hosts[${index}].cpu`}
+        label="CPU cores"
+        required
+        aria-required="true"
+      >
+        {/* Make this a masked input */}
+        <Select
+          id={`hosts[${index}].cpu`}
+          name={`hosts[${index}].cpu`}
+          options={[2, 4, 6, 8]}
+          {...rest}
+        />
+      </FormField>
+    </Box>
+  ),
+  memory: ({ key, index, ...rest }) => (
+    <Box key={key} width="xsmall">
+      <FormField
+        key={key}
+        htmlFor={`hosts[${index}].memory`}
+        name={`hosts[${index}].memory`}
+        label="Memory (GB)"
+        required
+        aria-required="true"
+      >
+        {/* Make this a masked input */}
+        <Select
+          id={`hosts[${index}].memory`}
+          name={`hosts[${index}].memory`}
+          options={['32', '64', '128', '256', '512']}
+          {...rest}
+        />
+      </FormField>
+    </Box>
+  ),
+};
+
+export const CreateCluster = () => {
+  const [formValues, setFormValues] = useState({
+    'cluster-name': '',
+    'resource-manager': false,
+    'automation-level': '',
+    'migration-threshold': 15,
+    hosts: [{ ...hostTemplate }],
```

Comment:
```suggestion
    hosts: [hostTemplate],
```
is this the same?

Reply:
I believe this would retain a pointer reference to the original hostTemplate, so any value updates made to it would update the reference. When any other "Add hosts" actions occur, they would be referencing the same object and its values. Right?

Would it be more clear if we did something like:
```
class Host {
  constructor() {
    this.name = '';
    this.host = '';
    this.cpu = '';
    this.memory = '';
  }
}

...

new Host();
```

instead?

Reply:
Thank you for clarifying, you're correct. I'm okay with your initial approach. The class structure is a bit more clear given the `new Host()` language, but also may be a bit verbose and less aligned to how we generally do things. I'm fine leaving as is.

Reply:
okay, leaving as is, but am open to the class approach for clarity in the future.